FEXCore/Source/Utils/FutexSpinWait.h

```diff
@@ -0,0 +1,282 @@
+#include <atomic>
+#include <chrono>
+#include <mutex>
+#include <type_traits>
+
+namespace FEXCore::Utils {
+#define SPINLOOP_8BIT " \
+  /* Prime the exclusive monitor with the passed in address. */ \
+  ldaxrb %w[Tmp], [%[Futex]]; \
+  /* WFE will wait for either the memory to change or spurious wake-up. */ \
+  wfe; \
+  /* Load with acquire to get the result of memory. */ \
+  ldarb %w[Result], [%[Futex]]; \
+"
```

Comment:
Why not have a single macro that takes `ldaxrb` and `ldarb` as arguments? That's less explicit than having the `_8BIT` suffix, but since the macros are only used in `sizeof(T)==1` branches the meaning is clear anyway.